Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Darwin.swift

```diff
@@ -0,0 +1,232 @@
+            guard result == 0 else {
+                try self.cleanupAll(input: input, output: output, error: error)
+                throw POSIXError(.init(rawValue: result) ?? .ENODEV)
+            }
+        }
+        // Setup spawnAttributes
+        posix_spawnattr_init(&spawnAttributes)
+        defer {
+            posix_spawnattr_destroy(&spawnAttributes)
+        }
+        var noSignals = sigset_t()
+        var allSignals = sigset_t()
+        sigemptyset(&noSignals)
+        sigfillset(&allSignals)
+        posix_spawnattr_setsigmask(&spawnAttributes, &noSignals)
+        posix_spawnattr_setsigdefault(&spawnAttributes, &allSignals)
+        // Configure spawnattr
+        var flags: Int32 = POSIX_SPAWN_CLOEXEC_DEFAULT |
+            POSIX_SPAWN_SETSIGMASK | POSIX_SPAWN_SETSIGDEF
+        if self.platformOptions.createProcessGroup {
+            flags |= POSIX_SPAWN_SETPGROUP
+        }
+        var spawnAttributeError = posix_spawnattr_setflags(&spawnAttributes, Int16(flags))
+        // Set QualityOfService
+        // spanattr_qos seems to only accept `QOS_CLASS_UTILITY` or `QOS_CLASS_BACKGROUND`
+        // and returns an error of `EINVAL` if anything else is provided
+        if spawnAttributeError == 0 && self.platformOptions.qualityOfService == .utility{
+            spawnAttributeError = posix_spawnattr_set_qos_class_np(&spawnAttributes, QOS_CLASS_UTILITY)
+        } else if spawnAttributeError == 0 && self.platformOptions.qualityOfService == .background {
+            spawnAttributeError = posix_spawnattr_set_qos_class_np(&spawnAttributes, QOS_CLASS_BACKGROUND)
+        }
+
+        // Setup cwd
+        var chdirError: Int32 = 0
+        if intendedWorkingDir != .currentWorkingDirectory {
+            chdirError = intendedWorkingDir.withPlatformString { workDir in
+                return posix_spawn_file_actions_addchdir_np(&fileActions, workDir)
+            }
+        }
+
+        // Error handling
+        if chdirError != 0 || spawnAttributeError != 0 {
+            try self.cleanupAll(input: input, output: output, error: error)
+            if spawnAttributeError != 0 {
+                throw POSIXError(.init(rawValue: result) ?? .ENODEV)
+            }
+
+            if chdirError != 0 {
+                throw CocoaError(.fileNoSuchFile, userInfo: [
+                    .debugDescriptionErrorKey: "Cannot failed to change the working directory to \(intendedWorkingDir) with errno \(chdirError)"
+                ])
+            }
+        }
+        // Run additional config
+        if let spawnConfig = self.platformOptions.additionalSpawnAttributeConfigurator {
+            try spawnConfig(&spawnAttributes)
+        }
+        if let fileAttributeConfig = self.platformOptions.additionalFileAttributeConfigurator {
+            try fileAttributeConfig(&fileActions)
+        }
+        // Spawn
+        var pid: pid_t = 0
+        let spawnError: CInt = executablePath.withCString { exePath in
```

Comment:
I suspect that this `withCString` should really be `withFileSystemRepresentation` because it is a path